osf/models/node.py

```diff
@@ -355,6 +355,30 @@ def parent_node(self):
                 return parent
         return None
 
+    @property
+    def quick_parent_id(self):
+        """
+        Optimization property. If node has been annotated with "annotated_parent_id"
+        in a queryset, use that value.  Otherwise, fetch the parent_node guid.
+        """
+        if hasattr(self, 'annotated_parent_id'):
+            return self.annotated_parent_id
+        else:
+            if self.parent_node:
+                return self.parent_node._id
+            return None
+
+    @property
+    def quick_tags(self):
+        """
+        Optimization property. If node has been annotated with "annotated_tags"
+        in a queryset, use that value.  Otherwise, fetch the parent_node guid.
+        """
+        if hasattr(self, 'annotated_tags'):
+            return [] if self.annotated_tags == [None] else self.annotated_tags
```

Comment:
`ArrayAgg`

> Returns a list of values, including nulls, concatenated into an array.

When annotating tags, if no tags, it returns [None] so I fix that here.